Tests Failing

```
$ ./tests.py
..F.F............
======================================================================
FAIL: test_blacklistTwoConnections (__main__.TSDBlacklistingTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests.py", line 86, in test_blacklistTwoConnections
    self.assertEqual(tsd1, (sender.host, sender.port))
AssertionError: Tuples differ: ('localhost', 4242) != ('localhost', 4243)

First differing element 1:
4242
4243

- ('localhost', 4242)
?                  ^

+ ('localhost', 4243)
?                  ^


======================================================================
FAIL: test_doublePickTwoConnections (__main__.TSDBlacklistingTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests.py", line 105, in test_doublePickTwoConnections
    self.assertEqual(tsd1, (sender.host, sender.port))
AssertionError: Tuples differ: ('localhost', 4242) != ('localhost', 4243)

First differing element 1:
4242
4243

- ('localhost', 4242)
?                  ^

+ ('localhost', 4243)
?                  ^


----------------------------------------------------------------------
Ran 17 tests in 0.015s

FAILED (failures=2)
```

--- a/tcollector.py
+++ b/tcollector.py
@@ -467,16 +467,11 @@ class SenderThread(threading.Thread):
             if hostport not in self.blacklisted_hosts:
                 break
         else:
-            if self.http is None:
-                LOG.info('No more healthy hosts, retry with previously blacklisted')
-                random.shuffle(self.hosts)
-                self.blacklisted_hosts.clear()
-                self.current_tsd = 0
-                hostport = self.hosts[self.current_tsd]
-            else:
-                # TODO: Enable multiple HTTP endpoints
-                LOG.debug('Reusing HTTP connection %s on port %s', self.host, self.port)
-                hostport = self.hosts[self.current_tsd]
+            LOG.info('No more healthy hosts, retry with previously blacklisted')
+            random.shuffle(self.hosts)
+            self.blacklisted_hosts.clear()
+            self.current_tsd = 0
+            hostport = self.hosts[self.current_tsd]
         self.host, self.port = hostport
         LOG.info('Selected connection: %s:%d', self.host, self.port)
 

